jkube-kit/config/image/src/main/java/org/eclipse/jkube/kit/config/image/GeneratorManager.java

```diff
@@ -16,6 +16,15 @@
 import java.util.List;
 
 public interface GeneratorManager {
+  /**
+   * Process provided set of ImageConfigurations. This processing may include filtering, merging with
+   * opinionated defaults and validating finally generated ImageConfigurations.
+   *
+   * @param imageConfigs list of ImageConfigurations to process
+   * @return processed list of ImageConfigurations
+   */
+  List<ImageConfiguration> generateAndMerge(List<ImageConfiguration> imageConfigs);
+
```

Comment:
Maybe we can actually remove the `generate` method because (from our internal meeting) it seems that this method is no longer relevant (generateAndMerge eventually calls generate)